src/LoopInterface.php

```diff
@@ -240,6 +240,42 @@ public function isTimerActive(TimerInterface $timer);
     public function futureTick(callable $listener);
 
     /**
+     * Registers a signal listener with the loop, which
+     * on it's turn registers it with a signal handler
+     * suitable for the loop implementation.
+     *
+     * See also [example #4](examples).
+     *
+     * @param int $signal
+     * @param callable $listener
+     *
+     * @throws UnsupportedFeatureException when signals
+     * aren't supported by the loop, e.g. when required
+     * extensions are missing.
+     *
+     * @return void
+     */
+    public function addSignal($signal, callable $listener);
+
+    /**
+     * Removed previous registered signal listener from
+     * the loop, which on it's turn removes it from the
+     * underlying signal handler.
+     *
+     * See also [example #4](examples).
+     *
+     * @param int $signal
+     * @param callable $listener
+     *
+     * @throws UnsupportedFeatureException when signals
+     * aren't supported by the loop, e.g. when required
+     * extensions are missing.
```

Comment:
Can this method really throw? Looks like adding isn't possible in this case either, so this actually boils down to removing a non-existent listener?

Reply:
I still think that this situation shouldn't throw. Removing a non-existent listener does not throw, so why would removing a listener throw when it's impossible to add listeners in the first place?

Reply:
Makes sense to me, they can catch the exception when trying top add it. I'll remove the exception on removing signal listeners :+1: